### PR TITLE
Fixes for n9k-f and n3k-f

### PR DESCRIPTION
**Pull request by saichint**
* This PR is for fixing beaker tests and software versioning scheme which changed from older versions to latest versions on n3k-f and n9k-f.
* Some of the features like vtp, acl, vxlan_vtp_vni etc. have a different behavior for these platforms and this is addressed.

All tests pass on n9k, n9k-f, n3k-f

### CODE REVIEW

File: examples/cisco/demo_vtp.pp [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
 
 class ciscopuppet::cisco::demo_vtp {
-  cisco_vtp { 'default':
-    ensure   => present,
-    domain   => 'cisco1234',
-    password => 'test1234',
-    version  => 2,
-    filename => 'bootflash:/vlan.dat'
+
+  if platform_get() =~ /n(3|5|6|7|8|9)k$/ {
```
Comment (mikewiebe) [inline examples/cisco/demo_vtp.pp]: We should be able to remove 8.

File: lib/puppet/parser/functions/platform_fretta.rb [hunk 2]
```diff
@@ -25,11 +25,13 @@ module Parser
     #
     module Functions
       newfunction(:platform_fretta, type: :rvalue) do |_args|
-        data = lookupvar('os')
+        data = lookupvar('cisco')
         return '' if data.nil?
 
-        pat = '7.0\(3\)F'
-        data['release']['full'].match(pat) ? true : false
+        data['inventory'].each do |_x, slot|
```
Comment (mikewiebe) [inline lib/puppet/parser/functions/platform_fretta.rb]: Did you verify the demo manifest with this change?

File: lib/puppet/parser/functions/platform_fretta.rb [hunk 3]
```diff
@@ -25,11 +25,13 @@ module Parser
...
+          return true if slot['pid'][/-R/]
```
Comment (mikewiebe) [inline lib/puppet/parser/functions/platform_fretta.rb]: In the `node_utils` function we check for both `FM-R` and `-R`.  Do we need both checks here or can we simplify it to just check `-R`?

File: lib/puppet_x/cisco/cmnutils.rb [hunk 4]
```diff
@@ -243,19 +243,27 @@ def self.range_summarize(range_str, sort=true)
         ranges.join(',').gsub('..', '-')
       end
 
+      # fretta check
+      def self.check_slot_pid(inv)
+        inv.each do |_x, slot|
+          return true if slot['pid'][/-R/]
```
Comment (mikewiebe) [inline lib/puppet_x/cisco/cmnutils.rb]: Same question here about `FM-R` vs `-R`

**Comment by mikewiebe** [comment]
@saichint Just a few comments.

File: lib/puppet/parser/functions/platform_fretta.rb (hunk 2 above)
Comment (saichint) [reply lib/puppet/parser/functions/platform_fretta.rb]: Yes

File: lib/puppet/parser/functions/platform_fretta.rb (hunk 3 above)
Comment (saichint) [reply lib/puppet/parser/functions/platform_fretta.rb]: Changed NU to use -R only

File: lib/puppet_x/cisco/cmnutils.rb (hunk 4 above)
Comment (saichint) [reply lib/puppet_x/cisco/cmnutils.rb]: Changed NU to use -R only

**Comment by mikewiebe** [comment]
👍

**Pull request merged**